VideoLocker/src/main/java/org/edx/mobile/model/course/BlockPath.java

```diff
@@ -0,0 +1,43 @@
+package org.edx.mobile.model.course;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Created by hanning on 6/1/15.
+ */
+public class BlockPath implements Serializable{
+    private List<CourseComponent> path = new ArrayList<>();
+
+    public void addPathNode(CourseComponent component){
```

Comment:
It's a little surprising that ``addPathNode`` does insert at beginning. Maybe that should be part of the name?